lib/formatter.ex

```diff
@@ -0,0 +1,289 @@
+      ~s|{"a":"b","c":2}|
+  """
+  @spec minimize(iodata, opts) :: binary
+  def minimize(iodata, opts \\ []) do
+    minimize_to_iolist(iodata, opts)
+    |> :erlang.list_to_binary
+  end
+
+
+  @doc ~S"""
+  Returns an iolist containing a minimized representation of
+  JSON-encoded `iodata`.
+
+  See `minimize/2` for details and options.
+  """
+  @spec minimize_to_iolist(iodata, opts) :: iolist
+  def minimize_to_iolist(iodata, opts) do
+    pretty_print_to_iolist(
+      iodata,
+      indent: "",
+      line_separator: "",
+      record_separator: opts[:record_separator] || "\n",
+      after_colon: ""
+    )
+  end
+
+
+  ## Returns a copy of `opts` with defaults applied
+  @spec normalize_opts(keyword) :: opts
+  defp normalize_opts(opts) do
+    [
+      indent: opts[:indent] || "  ",
+      line_separator: opts[:line_separator] || "\n",
+      record_separator: opts[:record_separator] || opts[:line_separator] || "\n",
+      after_colon: opts[:after_colon] || " ",
+    ]
+  end
+
+
+  ## Returns an iolist containing `depth` instances of `opts[:indent]`
+  @spec tab(opts, non_neg_integer, iolist) :: iolist
+  defp tab(opts, depth, output \\ []) do
+    if depth < 1 do
+      output
+    else
+      tab(opts, depth-1, [opts[:indent] | output])
+    end
+  end
+
+
+  @typep pp_state :: {
+    non_neg_integer,  ## depth -- current nesting depth
+    boolean,          ## in_str -- is the current byte in a string?
+    boolean,          ## in_bs -- does the current byte follow a backslash in a string?
+    boolean,          ## empty -- is the current object or array empty?
+    boolean,          ## first -- is this the first object or array in the input?
+  }
+
+  @spec pp_iodata(
+    iodata,           ## input -- input data
+    iolist,           ## output_acc -- output iolist (built in reverse order)
+    non_neg_integer,  ## depth -- current nesting depth
+    boolean,          ## in_str -- is the current byte in a string?
+    boolean,          ## in_bs -- does the current byte follow a backslash in a string?
+    boolean,          ## empty -- is the current object or array empty?
+    boolean,          ## first -- is this the first object or array in the input?
+    opts
+  ) :: {iolist, pp_state}
+  defp pp_iodata(input, output_acc, depth, in_str, in_bs, empty, first, opts)
+
+  defp pp_iodata("", output_acc, depth, in_str, in_bs, empty, first, opts) do
+    {:lists.reverse(output_acc), {depth, in_str, in_bs, empty, first, opts}}
+  end
+
+  defp pp_iodata([], output_acc, depth, in_str, in_bs, empty, first, opts) do
+    {:lists.reverse(output_acc), {depth, in_str, in_bs, empty, first, opts}}
+  end
+
+  defp pp_iodata(<<byte::size(8), rest::binary>>, output_acc, depth, in_str, in_bs, empty, first, opts) do
+    pp_byte(byte, rest, output_acc, depth, in_str, in_bs, empty, first, opts)
+  end
+
+  defp pp_iodata(byte, output_acc, depth, in_str, in_bs, empty, first, opts) when is_number(byte) do
```

Comment:
Let's use `is_integer` - a float doesn't make much sense in here.

Reply:
:+1: